zerver/views/user_settings.py

```diff
@@ -357,6 +374,84 @@ def json_change_settings(
     # TODO: Change the cache flushing strategy to make sure cache
     # does not contain stale objects.
     user_profile = UserProfile.objects.get(id=user_profile.id)
+
+    # Loop over user_profile.property_types
+    request_settings = {
+        setting_name: requested_value
+        for setting_name, requested_value in locals().items()
+        if setting_name in user_profile.property_types and requested_value is not None
+    }
+
+    if target_users is not None:
+        if not user_profile.is_realm_admin:
+            raise JsonableError(
+                _("Only organization administrators can change settings for other users.")
+            )
+
+        if timezone is not None:
+            raise JsonableError(
+                _("Organization administrators cannot change timezone setting for other users.")
+            )
+
+        for setting_name in request_settings:
+            if (
+                setting_name in UserBaseSettings.SECURITY_SENSITIVE_USER_SETTINGS
+                or setting_name in ["email", "new_password"]
+            ):
+                raise JsonableError(
+                    _(
+                        "Organization administrators cannot change sensitive settings for other users."
+                    )
+                )
+
+        user_ids = target_users.user_ids
+        group_member_ids = set()
+
+        if target_users.group_ids:
+            group_member_ids = set(
+                get_recursive_group_members_union_for_groups(target_users.group_ids).values_list(
+                    "id", flat=True
+                )
+            )
+
+        all_user_ids = list(set(user_ids) | group_member_ids)
+
+        # User settings do not apply to bots, so exclude them from the query even if they
+        # are in a member of a targeted group or are included in user_ids list. Deactivated
+        # users are included, so that the right thing happens if they are later reactivated.
+        users = user_ids_to_users(
+            all_user_ids, user_profile.realm, allow_deactivated=True, allow_bots=False
+        )
+
+        for setting_name, requested_value in request_settings.items():
+            if skip_if_already_edited:
+                already_edited_user_ids = set(
+                    RealmAuditLog.objects.filter(
+                        modified_user__in=users,
+                        event_type=AuditLogEventType.USER_SETTING_CHANGED,
+                        extra_data__property=setting_name,
+                    ).values_list("modified_user_id", flat=True)
```

Comment:
Here also we would do a separate query for each setting, but should be fine since we do not expect many settings to be updated in a single request too often.